osxkeychain/osxkeychain_darwin.go

```diff
@@ -21,6 +21,13 @@ import (
 // when the credentials are not in the keychain.
 const errCredentialsNotFound = "The specified item could not be found in the keychain."
 
+// errCredentialsNotFound is the specific error message returned by OS X
+// when environment does not allow showing dialog to unlock keychain.
+const errInteractionNotAllowed = "User interaction is not allowed."
+
+// ErrInteractionNotAllowed is returned if keychain password prompt can not be shown.
+var ErrInteractionNotAllowed = errors.New("keychain is locked for the current non-interactive session. Please run \"security -v unlock-keychain ~/Library/Keychains/login.keychain-db\" to unlock keychain and try again")
```

Comment:
@tonistiigi if the error doesn't necessarily mean that the keychain is locked, we should probably rephrase to something like `keychain may be locked ...`. And if you do change string literal use backticks to avoid ugly escaping.